src/array_string.rs

```diff
@@ -129,6 +129,28 @@ impl<const CAP: usize> ArrayString<CAP>
         Ok(vec)
     }
 
+    /// Creates a new `ArrayString` instance fully filled with ASCII NULL characters (`\0`). Useful
+    /// to be used as a buffer to collect external data or as a buffer for intermediate processing.
+    ///
+    /// ```
+    /// use arrayvec::ArrayString;
+    ///
+    /// let string = ArrayString::<16>::zero_filled();
+    /// assert_eq!(string.len(), 16);
+    /// ```
+    #[inline]
+    pub fn zero_filled() -> Self {
+        let mut vec = Self::new();
+        let data = [b'\0'; CAP];
+        // SAFETY: The array is fully filled with valid UTF-8
+        unsafe {
+            let ptr = data.as_ptr() as *const [MaybeUninit<u8>; CAP];
+            ptr.copy_to_nonoverlapping(&mut vec.xs as *mut _, 1);
+            vec.set_len(CAP);
+        }
+        vec
+    }
```

Comment:
I think we can and should copy the code in `fn new` verbatim, but instead of uninit, we use `MaybeUninit::zeroed` and len: CAP and it should be good

Reply:
Done!